x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/TypeConverter.java

```diff
@@ -297,13 +305,23 @@ private static String asString(Object nativeValue) {
     }
 
     private static Boolean asBoolean(Object val, EsType columnType, String typeString) throws SQLException {
-        return switch (columnType) {
-            case BOOLEAN, BYTE, SHORT, INTEGER, LONG, FLOAT, HALF_FLOAT, SCALED_FLOAT, DOUBLE -> Boolean.valueOf(
-                Integer.signum(((Number) val).intValue()) != 0
-            );
-            case KEYWORD, TEXT -> Boolean.valueOf((String) val);
-            default -> failConversion(val, columnType, typeString, Boolean.class);
-        };
+        switch (columnType) {
+            case BOOLEAN:
+            case BYTE:
+            case SHORT:
+            case INTEGER:
+            case LONG:
+            case FLOAT:
+            case HALF_FLOAT:
+            case SCALED_FLOAT:
+            case DOUBLE:
+                return Boolean.valueOf(Integer.signum(((Number) val).intValue()) != 0);
```

Comment:
This should be reverted to it's original `Boolean.valueOf(((Number) val).doubleValue() != 0)`, as it yields incorrect results for non-zero floats in (-1, 1)

Reply:
Hm, it had been set to `return Boolean.valueOf(Integer.signum(((Number) val).intValue()) != 0)`, before I started working on switch expression refactoring. Maybe it was changed at some point before the switch expression change?